antarest/storage/service.py

```diff
@@ -92,97 +105,116 @@ def export_study(
+            if self.study_service.check_errors(md)
             else StudyContentStatus.VALID
         )
-        self._save_metadata(uuid, params.user, content_status=status)
-        return uuid
+        self._save_metadata(md, owner=params.user, content_status=status)
+        return str(md.id)
 
     def import_output(
         self, uuid: str, stream: IO[bytes], params: RequestParameters
     ) -> JSON:
-        self._check_user_permission(params.user, uuid)
-        res = self.importer_service.import_output(uuid, stream)
+        md = self._get_metadata_according_permission(params.user, uuid)
+        res = self.importer_service.import_output(md, stream)
         return res
 
     def edit_study(
         self, route: str, new: JSON, params: RequestParameters
     ) -> JSON:
-        uuid, _, _ = self.study_service.extract_info_from_url(route)
-        self._check_user_permission(params.user, uuid)
-        return self.study_service.edit_study(route, new)
+        uuid, url = StorageServiceUtils.extract_info_from_url(route)
+        md = self._get_metadata_according_permission(params.user, uuid)
+        return self.study_service.edit_study(md, url, new)
 
     def _save_metadata(
         self,
-        uuid: str,
-        user: Optional[User],
+        metadata: Metadata,
+        owner: Optional[User] = None,
         content_status: StudyContentStatus = StudyContentStatus.VALID,
+        group: Optional[Group] = None,
     ) -> None:
-        if not user:
+        if not owner and not group:
             raise UserHasNotPermissionError
 
-        info = self.study_service.get_study_information(uuid)["antares"]
-        meta = Metadata(
-            id=uuid,
-            name=info["caption"],
-            version=info["version"],
-            author=info["author"],
-            created_at=datetime.fromtimestamp(info["created"]),
-            updated_at=datetime.fromtimestamp(info["lastsave"]),
-            content_status=content_status,
-            users=[user],
-        )
-        self.repository.save(meta)
+        info = self.study_service.get_study_information(metadata)["antares"]
+
+        metadata.name = info["caption"]
+        metadata.version = info["version"]
+        metadata.author = info["author"]
+        metadata.created_at = datetime.fromtimestamp(info["created"])
+        metadata.updated_at = datetime.fromtimestamp(info["lastsave"])
+        metadata.content_status = content_status
 
-    def _check_user_permission(
-        self, user: Optional[User], uuid: str, raising: bool = True
-    ) -> bool:
-        def check(user: Optional[User], uuid: str) -> None:
-            if not user:
-                raise UserHasNotPermissionError()
+        if owner:
+            metadata.owner = owner
+        if group:
+            metadata.groups = [group]
+        self.repository.save(metadata)
 
-            if user.role == Role.ADMIN:
-                return
+    def _get_metadata_according_permission(
```

Comment:
why not have a method to retrieve metadata (raise StudyNotFound if not found) and an other one (the previously named "check_user_permission" with user and metadata in input ?

Reply:
Flag change return expectation so method signature. When flag is off, method return a Metadata or send error. When flag is on, return can be None. Therefore I separate use in different method with their own return type